From 05420f45851ae85e1049ddc0d6dc79f0433c448c Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Fri, 26 May 2017 10:19:37 -0700
Subject: [PATCH 2468/2848] HDFS-7939. Two fsimage_rollback_* files are
 created which are not deleted after rollback.
 (Contributed by J.Andreina)

(cherry picked from commit 987c9e12e184b35a5abab49f4188e22509ad63a5)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt

(cherry picked from commit 4aba069b3793edef352662bd4106c0b33a1f0a3b)

Change-Id: I80a85deb9c9d53abb42bd64e9d79d8845e292c73
(cherry picked from commit 1c657a084566023e843eb6dbe9e2521161b313bc)
---
 .../hadoop/hdfs/server/namenode/FSImage.java       |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImage.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImage.java
index df93137..43b87ab 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImage.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImage.java
@@ -727,6 +727,8 @@ private void rollingRollback(long discardSegmentTxId, long ckptId)
         true);
     // purge all the checkpoints after the marker
     archivalManager.purgeCheckpoinsAfter(NameNodeFile.IMAGE, ckptId);
+    // HDFS-7939: purge all old fsimage_rollback_*
+    archivalManager.purgeCheckpoints(NameNodeFile.IMAGE_ROLLBACK);
     String nameserviceId = DFSUtil.getNamenodeNameServiceId(conf);
     if (HAUtil.isHAEnabled(conf, nameserviceId)) {
       // close the editlog since it is currently open for write
-- 
1.7.9.5

